src/Bridge/Symfony/Messenger/DataTransformer/InputMessageDataTransformer.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Messenger\DataTransformer;
+
+use ApiPlatform\Core\Bridge\Symfony\Messenger\InputMessage;
+use ApiPlatform\Core\DataTransformer\DataTransformerInterface;
+
+/**
+ * Transforms an input that implements the InputMessage interface
+ * to itself. This gives the ability to send the Input to a
+ * message handler and process it asynchronously.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class InputMessageDataTransformer implements DataTransformerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function transform($object, string $to, array $context = [])
+    {
+        return $object;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsTransformation($object, string $to, array $context = []): bool
```

Comment:
Not really related to this but I suggest changing `$object` with `$data` while it's still possible. It's a bit confusing.